### DISCUSSION TITLE
Protonmail bridge

### DISCUSSION THREAD

**Discussion by joskapista44** [nextcloud/all-in-one#5150] [Ideas] [answered]
If you want to use protonmail accounts in mail, it is not possible now.
There is only a complicated solution to install and setup it but is only for professionals. 
A solution for normal would be great: integrate protonmail bridge to nextcloud as an app with a web based setup interface could be prefect to everyone.

**Comment by szaimen** [comment, marked as answer]
Hi, I just wanted to mention that we have this now: https://github.com/nextcloud/all-in-one/tree/main/community-containers#how-to-add-containers. So the community could potentially add these containers as additional container. Feel free to ping me if you should need help on this!  

I found this container image which could potentially be used here: https://hub.docker.com/r/shenxn/protonmail-bridge

**Reply by szaimen** [reply]
Regarding the json, I just leave this here as an example:

```json
{
    "aio_services_v1": [
        {
            "container_name": "nextcloud-aio-protonmail-bridge",
            "display_name": "Protonmail Bridge",
            "documentation": "https://github.com/nextcloud/all-in-one/tree/main/community-containers/protonmail-bridge",
            "image": "shenxn/protonmail-bridge",
            "image_tag": "build",
            "internal_port": "143",
            "restart": "unless-stopped",
            "environment": [
                "TZ=%TIMEZONE%"
            ],
            "volumes": [
                {
                    "source": "nextcloud_aio_protonmail_bridge",
                    "destination": "/root",
                    "writeable": true
                }
            ],
            "backup_volumes": [
                "nextcloud_aio_protonmail_bridge"
            ]
        }
    ]
}
```

**Reply by szaimen** [reply]
Hi @joskapista44  since you are very interested in this feature IIRC, would you be up for creating the PR that adds this?

**Reply by kravitsjacob** [reply]
Hello!

Did anything ever come of this? I would like to do the same thing - use proton mail with nextcloud. Given the similarities of the philosophies of the two projects, I was excited to see people already discussing it!

I tried configuring the protonmail-bridge docker that was linked above, but I got an "IMAP server is not reachable" error from nextcloud. Perhaps there are some ports I'm not exposing properly - I'm new to both protonmail and nextcloud...

**Reply by Fuseteam** [reply]
this is on the list of https://github.com/nextcloud/all-in-one/issues/5251 where contributors are wanted

**Reply by szaimen** [reply]
@Fuseteam would you be up for creating the PR that adds this?

**Reply by Fuseteam** [reply]
@szaimen I don't actually have a valid subscription to use the proton mail bridge

**Reply by szaimen** [reply]
@kravitsjacob would you be up for creating the PR that adds this?

**Reply by kravitsjacob** [reply]
Hello @szaimen! I was not planning on creating a PR for this as a solution didn’t require any code. I have been using Nextcloud with protonmail for quite some time after following [these instructions](https://github.com/nextcloud/all-in-one/discussions/6154#discussioncomment-12487371). It seems like others were able to follow as well!

**Reply by szaimen** [reply]
I see. The idea was to add this kind of config as a community container so that others can easily use it. See https://nextcloud.com/blog/how-to-use-nextcloud-aio-using-community-containers/

**Reply by Fuseteam** [reply]
yeah the idea is to pre-configure it, so that it is even less steps for users

**Reply by kravitsjacob** [reply]
Ahh I gotcha. So the user installs some sort of `protonmail-setup` Nextcloud container which would allow them to do the install and setup process through Nextcloud AIO instead of having to do it manually?

I could potentially be interested, but I’m pretty slammed with my day job, so it’s been hard to find motivation to come home and do more coding, haha!

Is there any documentation for container _maintainers_? I see a lot of docs on how to _use_ containers but I don’t see as much on best practices for container _maintainers_. Doesn’t seem like a crazy complicated setup to make a container, but was just wondering if there was anything out there!

**Reply by szaimen** [reply]
See https://github.com/nextcloud/all-in-one/tree/main/community-containers#how-to-add-containers

**Reply by szaimen** [reply]
Also see https://github.com/nextcloud/all-in-one/discussions/5150#discussioncomment-10488702

```json
{
    "aio_services_v1": [
        {
            "container_name": "nextcloud-aio-protonmail-bridge",
            "display_name": "Protonmail Bridge",
            "documentation": "https://github.com/nextcloud/all-in-one/tree/main/community-containers/protonmail-bridge",
            "image": "shenxn/protonmail-bridge",
            "image_tag": "build",
            "internal_port": "143",
            "restart": "unless-stopped",
            "environment": [
                "TZ=%TIMEZONE%"
            ],
            "volumes": [
                {
                    "source": "nextcloud_aio_protonmail_bridge",
                    "destination": "/root",
                    "writeable": true
                }
            ],
            "backup_volumes": [
                "nextcloud_aio_protonmail_bridge"
            ]
        }
    ]
}
```

**Reply by szaimen** [reply]
@kravitsjacob please let me know if you need any help on this or if you have questions :)

**Reply by DenuxPlays** [reply]
Btw proton enabled their SMTP Server for non buisness users a while ago.
I dropped the mail bridge in favour of it.

**Reply by gggeek** [reply]
@szaimen does/could the MC push the smtp-server config to NC (and to all other containers which do send emails)? If this is the case, it would be relatively straightforward to add to the AIO stack a container dedicated to eg. debug emails sent, useful during development, such as maildev or mailcatcher...

**Reply by gggeek** [reply]
Answering to myself above: it seems that yes, community containers can push config to the NC instance by using lines such as `php occ config:system:set mail_smtpport ...` under the `nextcloud_exec_commands` key in their json definition file

**Reply by szaimen** [reply]
Hi @gggeek did you plean to work on this?

**Reply by gggeek** [reply]
@szaimen it's way down in my todo list, so not in the near term, sadly...
